src/price_data.py

```diff
@@ -326,3 +328,133 @@ def get_cost(self, tr: Union[Operation, SoldCoin], reference_coin: str = config.
         if isinstance(tr, SoldCoin):
             return price * tr.sold
         raise NotImplementedError
+    
+    def get_candles(self, start: int, stop: int, symbol: str) ->list:
+        if self.exchange.has['fetchOHLCV']:
+            sleep(self.exchange.rateLimit / 1000)  # time.sleep wants seconds
+            # get 2min before and after range
+            return self.exchange.fetch_ohlcv(symbol, '1m', start-1000*60*2, max(int((stop-start)/1000/60)+5, 1))
+        else:
+            logging.warning(
+                "fetchOHLCV not implemented on exchange, skipping priceloading using ohlcv")
+            raise None
+
+    def initialize_ccxt(self):
```

Comment:
initialize_ccxt can take a while especially on exchanges with large paircount (binance is at around 1k) and a slow internet connection. thats why i only call it once in init.
but in theory loading all exchange pair data for all used exchanges could easily be done upfront if there is a way to find out which exchanges we need upfront

Reply:
If you want to initialize them beforehand, I think that we should move the class variables to an `Price.Data__init__` function like.. and it might be nice to have an AssetPair namedtuple, which stores our tuple?

```python
def __init__(self):
  self.exchanges: dict[str, AssetPair] = {}
  ...
```

We could call something like `price_data.initialize(platforms)` after the read_in in `main.py`.

Reply:
this is also obsolete, due to the graph approach